aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,271 @@
+---
+title: Run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop
+author: pavelsavara
+description: Learn how to run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop.
+monikerRange: '>= aspnetcore-7.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 10/18/2022
+uid: client-side/import-export-interop
+---
+# Run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop
+
+This article explains how to run .NET from JavaScript (JS) using `[JSImport]`/`[JSExport]` interop.
+
+Existing JS apps can use the expanded client-side WebAssembly support in .NET 7 to reuse .NET libraries from JS or to build novel .NET-based apps and frameworks.
+
+> [!NOTE]
+> This article focuses on running .NET from JS apps without any dependency on [Blazor](xref:blazor/index). 
+
+<!--
+
+    HOLD: ADD TO NOTE AFTER BLAZOR ARTICLE IS MERGED ...
+
+    For guidance on using `[JSImport]`/`[JSExport]` interop in Blazor WebAssembly apps, see <xref:blazor/js-interop/import-export-interop>.
+
+-->
+
+These approaches are appropriate when you only expect to run on WebAssembly (:::no-loc text="WASM":::). Libraries can make a runtime check to determine if the app is running on :::no-loc text="WASM"::: by calling <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType>.
+
+## Prerequisites
```

Comment:
To use .NET from JS using this model you need to:

- Install the `wasm-tools` workload, which brings in the related MSBuild targets (you only need the `wasm-experimental` workload if you want to use the project templates)
- Target `net7.0` or later
- Specify `browser-wasm` for the runtime identifier: `<RuntimeIdentifier>browser-wasm</RuntimeIdentifier>`
- Specify `<OutputType>Exe</OutputType>` (we may support library projects without an entry point later)
- Enable `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` (you've already got this covered below, but I'm listing it here for completeness)

Currently, you also need to specify `WasmMainJSPath` to point to some file on disk. This file will get published with the app, but it doesn't actually have to be used if you're going to integrate .NET into an existing JS app. We might make this property optional in the future: https://github.com/dotnet/runtime/issues/77191.